src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -118,46 +112,43 @@ private LiKafkaConsumerImpl(LiKafkaConsumerConfig configs,
 
-    // Instantiate segment deserializer if needed.
-    Deserializer segmentDeserializer = largeMessageSegmentDeserializer != null ? largeMessageSegmentDeserializer :
-        configs.getConfiguredInstance(LiKafkaConsumerConfig.SEGMENT_DESERIALIZER_CLASS_CONFIG, Deserializer.class);
-    segmentDeserializer.configure(configs.originals(), false);
-
     // Instantiate message assembler if needed.
     int messageAssemblerCapacity = configs.getInt(LiKafkaConsumerConfig.MESSAGE_ASSEMBLER_BUFFER_CAPACITY_CONFIG);
     int messageAssemblerExpirationOffsetGap = configs.getInt(LiKafkaConsumerConfig.MESSAGE_ASSEMBLER_EXPIRATION_OFFSET_GAP_CONFIG);
     boolean exceptionOnMessageDropped = configs.getBoolean(LiKafkaConsumerConfig.EXCEPTION_ON_MESSAGE_DROPPED_CONFIG);
     MessageAssembler assembler = new MessageAssemblerImpl(messageAssemblerCapacity, messageAssemblerExpirationOffsetGap,
-                                                          exceptionOnMessageDropped, segmentDeserializer);
+                                                          exceptionOnMessageDropped);
 
     // Instantiate delivered message offset tracker if needed.
     int maxTrackedMessagesPerPartition = configs.getInt(LiKafkaConsumerConfig.MAX_TRACKED_MESSAGES_PER_PARTITION_CONFIG);
     DeliveredMessageOffsetTracker messageOffsetTracker = new DeliveredMessageOffsetTracker(maxTrackedMessagesPerPartition);
 
     // Instantiate auditor if needed.
-    Auditor<K, V> auditor = consumerAuditor != null ? consumerAuditor :
+    _auditor = consumerAuditor != null ? consumerAuditor :
         configs.getConfiguredInstance(LiKafkaConsumerConfig.AUDITOR_CLASS_CONFIG, Auditor.class);
-    auditor.configure(configs.originals());
-    auditor.start();
+    _auditor.configure(configs.originals());
+    _auditor.start();
 
     // Instantiate key and value deserializer if needed.
-    Deserializer<K> kDeserializer = keyDeserializer != null ? keyDeserializer :
+    _keyDeserializer = keyDeserializer != null ? keyDeserializer :
         configs.getConfiguredInstance(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, Deserializer.class);
-    kDeserializer.configure(configs.originals(), true);
-    Deserializer<V> vDeserializer = valueDeserializer != null ? valueDeserializer :
+    _keyDeserializer.configure(configs.originals(), true);
+    _valueDeserializer = valueDeserializer != null ? valueDeserializer :
         configs.getConfiguredInstance(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, Deserializer.class);
-    vDeserializer.configure(configs.originals(), false);
+    _valueDeserializer.configure(configs.originals(), false);
 
     // Instantiate consumer record processor
-    _consumerRecordsProcessor = new ConsumerRecordsProcessor<>(assembler, kDeserializer, vDeserializer,
-                                                               messageOffsetTracker, auditor);
+    _consumerRecordsProcessor = new ConsumerRecordsProcessor(assembler, messageOffsetTracker);
```

Comment:
I would rather hide the ConsumerRecordsProcessor and not let people reuse it.  That being said resuse is even better when the CRP just does one thing like assemble large messages (or pass messages through) and do the necessary offset tracking.  If we (or someone else) has their own clients and has plugins (or something like them) then they can integrate the CRP into their record processing flow where it is needed.  In our case we may want to have plugins that run before or after deserialization and before or after large message support.  Having large message also do audit and deserialization conflates the various purposes of the CRP.